ownCloudAppShared/Client/View Controllers/Location Picker/ClientLocationPicker.swift

```diff
@@ -172,6 +172,9 @@ public class ClientLocationPicker : NSObject {
 			}
 
 			allowedLocationFilter = { (targetLocation, context) in
+				OnMainThread {
+					self.headerView?.secureView(core: context?.core)
```

Comment:
The use of `self` here might create a retain-loop.
Also view management code shouldn't be placed into this closure, if only because it could be called more than once and a new watermark view would be added on every call.